feat: Introduce scrape label for usage in NetworkPolicy (#1557)
kyma-project/telemetry-manager: docs/user/04-metrics.md

**Remedy 2**: Define the application protocol in the Service port definition by either prefixing the port name with the protocol, like in `http-metrics` or define the `appProtocol` attribute.

> **Cause 3**: A deny all `NetworkPolicy` was created in the workload namespace and the agent cannot scrape metric from annotated workloads.

Reviewer: ```suggestion
**Cause 3**: A deny-all `NetworkPolicy` was created in the workload namespace, which prevents that the agent can scrape metrics from annotated workloads.
```
If I understand this correctly :)

> **Remedy 3**: Define a `NetworkPolicy` to let the agent scrape your workload using the `telemetry.kyma-project.io/metric-scrape` label.

Reviewer: considering that the cause says that a network policy already exists, why does creating a network policy solve the problem?
Are we suggesting to replace the deny-all policy with one that's more permissive?

```suggestion
**Remedy 3**: Replace the deny-all `NetworkPolicy` with one that allows the agent to scrape your workload using the `telemetry.kyma-project.io/metric-scrape` label.
```

Author: technically, it's possible to modify the deny-all policy to be more permissive, but a more common practice is to define another network policy on top of the deny-all policy, explicitly allowing communication between selected pods, this leaves the deny-all policy untouched and allows for more additional policies to be added separately, but I'll discuss with Icke tomorrow just to be sure that it's what we're going for.

Reviewer: I see, thank you!
